### ISSUE TITLE
MO2 does not work with flatpak

### ISSUE THREAD

**Issue by AZCurt** [rockerbacon/modorganizer2-linux-installer#289]
This install is on Manjaro Linux, with Protontricks installed via the Manjaro (Arch) flatpak. Upon running the install.sh script, I get the following error.  **ERROR: missing dependencies protontricks**

I have tried using the alias from the Protontricks github page;
**echo "alias protontricks='flatpak run com.github.Matoking.protontricks'" >> ~/.bashrc**

I receive the same error whether the alias is present, or removed with unalias and from ./bashrc

However running the check command from within check_dependencies.sh (command -v protontricks) I get the following. with the alias present   **alias protontricks='flatpak run com.github.Matoking.protontricks'**

Without the alias it returns nothing.

Protontricks Steam library locations have been updated, and all available games are visible when ran from the protontricks applications menu. It is highly likely I missed something simple, but one never knows.

**Comment by rockerbacon** [comment]
`check_dependencies.sh` really only checks if `command -v protontricks` outputs something. There are no dependency checks after that either. If you get any output after running `command -v protontricks` in your terminal, the script should not have had any issues.

Things you can try:
1. Ensure you close and reopen your terminal after adding the alias to .bashrc, your terminal only loads .bashrc at startup;
2. Ensure the installer is being executed with bash and not any other shell interpreter. You can do so with the command `bash install.sh`;
3. Ensure the installer is being executed by the same user for which the alias was added;
4. Manually create the alias right before running the installer. You can do that by executing `alias protontricks='flatpak run com.github.Matoking.protontricks'` in your terminal

**Comment by rockerbacon** [comment]
You can also try to disable the check if all fails. Just prepend a `#` to line 55 of `install.sh`

**Comment by AZCurt** [comment]
Thank you for the fast reply. It came down to commenting out line 55. However during install the following error came up involving protontricks again. 
INFO: installing nxm link broker in '~/.local/share/modorganizer2'
INFO: installing nxm link handler in '~/.local/share/applications/'
INFO: applying protontricks xaudio2_7=native
~/mo2-installer/step/configure_steam_wineprefix.sh: line 5: protontricks: command not found

I will start this all over from scratch just in case protontricks, etc did not install correctly initially. Double check paths, ownership and the like. I just find it odd that every indication shows it being installed and functioning, but one never knows. I will update this once I have gone thought everything.

**Comment by AZCurt** [comment]
Not sure why I keep getting the protontricks error. Paths, permissions, user etc all check out the best I can see. I can even start MO2 within the install directory and protontricks asks which Wine prefix I want to use.

![Protontricks](https://user-images.githubusercontent.com/44403834/149046197-a563dd39-3b73-4189-adc4-f63f49b2330d.png)

Unless there is anything I missed, or other suggestions to try, I will close this tomorrow.

Thanks!

**Comment by rockerbacon** [comment]
@AZCurt I researched a bit and aliases are exclusively for when the shell is running in interactive mode, they won't be expanded when running a script.

A quick solution for you is to create a small file named `protontricks` and place it in a directory included in your path:
```sh
#!/bin/bash
flatpak run 'flatpak run com.github.Matoking.protontricks' "$@"
```

Don't forget to make the file executable with `chmod +x /path/to/protontricks`.
You can see all directories in your path with `echo "$PATH"`. I'd recommend using `~/.local/bin` if you have it.

**Comment by rockerbacon** [comment]
Support for the flatpak version of protontricks will require additional development.

**Comment by rockerbacon** [comment]
@AZCurt could you check if pre-release [4.2.1](https://github.com/rockerbacon/modorganizer2-linux-installer/releases/tag/4.2.1) solves your issue?

**Comment by AZCurt** [comment]
Will check it now and let you know.

**Comment by AZCurt** [comment]
Error applying Protontricks to xaudio2_7 with the following message. Everything else ran with no problem. 

error: Invalid id flatpak run com.github.Matoking.protontricks: Name can't contain

**Comment by AZCurt** [comment]
Here is the output prior to the error. I should have added it initially as a better guide.

INFO: installing nxm link broker in '/home/curt/.local/share/modorganizer2'
INFO: installing nxm link handler in '/home/curt/.local/share/applications/'
INFO: applying protontricks xaudio2_7=native
error: Invalid id flatpak run com.github.Matoking.protontricks: Name can't contain

**Comment by AZCurt** [comment]
More info. If I remove the protontricks fine I created and run it normally I get the following error. I do have Winetricks installed from the Official Repositories and it is available from the command line.

INFO: installing nxm link broker in '/home/curt/.local/share/modorganizer2'
INFO: installing nxm link handler in '/home/curt/.local/share/applications/'
INFO: applying protontricks xaudio2_7=native
protontricks (ERROR): The WINETRICKS path is invalid, please make sure Winetricks is installed in that path!
Winetricks isn't installed, please install winetricks in order to use this script!

**Comment by rockerbacon** [comment]
Try [4.2.2](https://github.com/rockerbacon/modorganizer2-linux-installer/releases/tag/4.2.2) now

**Comment by AZCurt** [comment]
I received the same errors using the flatpak version of protontricks as before. Prior to testing I did the following.
Reinstalled wine, winetricks, wine-mono and wine_gecko. Just in case there was something there.

Having failed, I installed 4.2.0 with the AUR repository version, just to see if something was amiss with my install. MO 2 installed and ran successfully from within Steam. This validated the environment I believe. I will remove the install, etc and continue to test the flatpak version if you wish.

**Comment by rockerbacon** [comment]
Just a small update: only the flatpak version of protontricks works with the flatpak version of Steam. This means this issue keeps the installer from working with the flatpak version of Steam as well;

**Comment by rockerbacon** [comment]
[4.2.3](https://github.com/rockerbacon/modorganizer2-linux-installer/releases/tag/4.2.3) should fix the winetricks path error.

**Comment by AZCurt** [comment]
Removed the AUR version of protontricks, rebooted & installed the Flatpac version, set paths etc. Updated the existing install with 4.2.3 with no problems. Rebooted and re-installed as update. Again installed without problems. I have attached my install log for review. 

One thing though, I had to change the proton version to Proton-Experimental for MO2 and ultimately SSE to run. I believe this was changed by a recent update to SSE from me initially installing this with the AUR protontricks version. 

Thanks for the update!

[InstallLog.txt](https://github.com/rockerbacon/modorganizer2-linux-installer/files/7945274/InstallLog.txt)
